src/plugins/intel_gpu/include/intel_gpu/runtime/layout.hpp

```diff
@@ -126,120 +128,124 @@ inline data_types element_type_to_data_type(ov::element::Type t) {
     }
 }
 
+constexpr size_t SHAPE_RANK_MAX = 9;
```

Comment:
layout::max_rank() returns SHAPE_RANK_MAX, that's okay for sure.

However for now I have to leave SHAPE_RANK_MAX  9 instead of 8, as tensor uses 9, and there are tons of padding objects initiated from tensor.sizes(), etc.

We would change it to 8 once we clear up tensor. How do you think?

Reply:
Unfortunately, some extra and tedious workaround probably in need to align shape and padding with 9, as layout::max_rank() is used by primitive_inst::fill_shape_info_data, which is inconsistent to the underlying kernel selector impl I think. DataTensor::max_rank() returns 8.

So the best way I think is to let it mismatch between shape and padding and clear up tensor class asap. So far it won't arouse a a problem, as this mismatch is handled in layout::get_padded_dims().